### PR TITLE
Check for sufficient staked ANA to borrow NIRV

### PR DESCRIPTION
**Pull request by diegowilson**

### CODE REVIEW

File: programs/nirvana/src/state/user_reward.rs [hunk 1]
```diff
@@ -16,6 +16,9 @@ pub struct UserReward {
     /// How much ANA is staked
     pub staked_amount: ANA,
 
+    /// How much NIRV is borrowed
+    pub borrowed_nirv: NIRV,
```
Comment (the-fool) [inline programs/nirvana/src/state/user_reward.rs:20]: nice call.
At first I was thinking of storing the NIRV debt in the `history` account -- but this seems like a better home.

File: programs/nirvana/src/instructions/borrow_nirv.rs [hunk 2]
```diff
@@ -20,15 +26,47 @@ pub struct BorrowNirv<'info> {
...
+
+    max_nirv_balance.rescale(nirv_decimals);
+
+    let borrowed_nirv = ctx.accounts.user_reward_index.borrowed_nirv.to_decimal();
+    let new_nirv_loan_d = NIRV::from_u64(new_nirv_loan).to_decimal();
+    let borrowed_nirv_after_loan = borrowed_nirv.add(new_nirv_loan_d);
+    if borrowed_nirv_after_loan > max_nirv_balance {
+        return Err(ErrorCode::InsufficientStakedANAToBorrowNIRV.into());
+    }
+    ctx.accounts.user_reward_index.borrow_nirv(NIRV::from_u64(new_nirv_loan));
```
Comment (the-fool) [inline programs/nirvana/src/instructions/borrow_nirv.rs:64]: If you move this line up, you can skip the arithmetic on line 60.

```rust
ctx.accounts.user_reward_index.borrow_nirv(NIRV::from_u64(new_nirv_loan));
if ctx.accounts.user_reward_index.borrowed_nirv > max_nirv_balance.mantissa().try_into().unwrap() {
   . . . 
}
```

**Review by the-fool — COMMENTED** [review]
easy as that

File: programs/nirvana/src/instructions/borrow_nirv.rs (hunk 2 above)
Comment (diegowilson) [reply programs/nirvana/src/instructions/borrow_nirv.rs:64]: Good catch https://github.com/nirvanadao/program/pull/279

**Pull request merged**